src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py

```diff
@@ -930,50 +929,66 @@ def get_test_desc(self, test: str) -> str:
 
-        Use the following environment variables:
+        This method use the following environment variables:
 
          - LINUX_DUT_IP
             * if not provided, the Matter app is assumed to run on the same machine as the test,
               such as during CI, and the commands are sent to it using a local named pipe
             * if provided, the commands for writing to the named pipe are forwarded to the DUT
         - LINUX_DUT_USER
-
             * if LINUX_DUT_IP is provided, use this for the DUT user name
             * If a remote password is needed, set up ssh keys to ensure that this script can log in to the DUT without a password:
                  + Step 1: If you do not have a key, create one using ssh-keygen
                  + Step 2: Authorize this key on the remote host: run ssh-copy-id user@ip once, using your password
                  + Step 3: From now on ssh user@ip will no longer ask for your password
+
         """
 
-        if app_pipe_name is None:
-            app_pipe_name = self.get_default_app_pipe_name()
+        if app_pipe_prefix is None:
+            app_pipe_prefix = self.matter_test_config.app_pipe_prefix
+        if app_pid is None:
+            app_pid = self.matter_test_config.app_pid
 
-        if not isinstance(app_pipe_name, str):
-            raise TypeError("the named pipe must be provided as a string value")
+        if not isinstance(app_pipe_prefix, str):
+            raise TypeError("The named pipe must be provided as a string value")
 
         if not isinstance(command_dict, dict):
-            raise TypeError("the command must be passed as a dictionary value")
+            raise TypeError("The command must be passed as a dictionary value")
 
         import json
         command = json.dumps(command_dict)
 
         import os
         dut_ip = os.getenv('LINUX_DUT_IP')
 
+        # Checks for concatenate app_pipe and app_pid
+        if not isinstance(app_pid, int):
+            raise TypeError("The --app-pid flag is not instance of int")
+        # Verify we have a valid app-id
+        if app_pid == 0:
+            asserts.fail("app_pid is 0 , is the flag --app-pid set?. app-id flag must be set in order to write to pipe.")
+        app_pipe_name = app_pipe_prefix + str(app_pid)
+
         if dut_ip is None:
             if not os.path.exists(app_pipe_name):
                 # Named pipes are unique, so we MUST have consistent PID/paths
-                # set up for them to work.
+                # Set up for them to work.
                 logging.error("Named pipe %r does NOT exist" % app_pipe_name)
                 raise FileNotFoundError("CANNOT FIND %r" % app_pipe_name)
             with open(app_pipe_name, "w") as app_pipe:
+                logger.info(f"Sending out-of-band command: {command} to file: {app_pipe_name}")
                 app_pipe.write(command + "\n")
             # TODO(#31239): remove the need for sleep
             sleep(0.001)
```

Comment:
feedback on code below if we touch it (cannot add inline): we should use shlex to escape not `command.replace(....`.

Somewhere in there we should have `command_fixed = shelx.quote(json.dumps(command))`